lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -418,6 +425,62 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     return success();
   }
 
+  LogicalResult handleTopModuleRefPorts() {
```

Comment:
All ref ports are already removed in the `garbageCollect` function that's called after this.

I suppose you could do it for the top module here, although if we expand this to support multiple "public" modules in a circuit, it wouldn't be safe to start deleting ref ports here in case one public ref port has dataflow to another one in a separate module.

Reply:
we do need to do all public modules.

Reply:
I don't care were they are deleted, as long as they are.